### PR TITLE
Adds ability to synchronously load page objects.

### PR DESCRIPTION
**Pull request by staats-google**

### CODE REVIEW

**Comment by staats-google** [comment]
Ignore this for now. I'm having some trouble with the tests and want to run it on another machine.

File: dart/async/lib/html.dart [hunk 1]
```diff
@@ -50,15 +50,22 @@ class HtmlPageLoader extends BasePageLoader {
...
+  dynamic _getContext(dynamic context) {
     if (context != null) {
       if (context is Node) {
         context = new HtmlPageLoaderElement(context, this);
       } else if (context is! HtmlPageLoaderElement) {
         throw new PageLoaderException('Invalid context: $context');
       }
     }
-    return getInstanceInternal(type, context);
+    return null;
```
Comment (manuelF) [inline dart/async/lib/html.dart]: Shouldn't it be return context?

File: dart/async/lib/webdriver.dart [hunk 2]
```diff
@@ -66,15 +66,22 @@ class WebDriverPageLoader extends BasePageLoader {
...
+  WebDriverPageLoaderElement _getContext(dynamic context) {
     if (context != null) {
       if (context is async_wd.SearchContext) {
         context = new _BaseWebDriverPageLoaderElementAsync(context, this);
       } else if (context is! _BaseWebDriverPageLoaderElementAsync) {
         throw new PageLoaderException('Invalid context: $context');
       }
     }
-    return getInstanceInternal(type, context);
+    return null;
```
Comment (manuelF) [inline dart/async/lib/webdriver.dart]: Again, why null? You have a context already.

File: dart/async/lib/webdriver_sync.dart [hunk 3]
```diff
@@ -256,27 +288,27 @@ class _WebDriverPageLoaderElementSync
         super._(loader, properties: new _DocumentPropertiesSync(context));
 
   @override
-  Future<String> get name => throw _unsupportedAsyncException;
+  Future<String> get name async => nameSync;
```
Comment (manuelF) [inline dart/async/lib/webdriver_sync.dart]: Shouldn't it be 
Future<String> get name async => new Future.value(nameSync) ?

File: dart/async/lib/html.dart (hunk 1 above)
Comment (staats-google) [inline dart/async/lib/html.dart]: Whoa, hahah what a slip.

File: dart/async/lib/webdriver_sync.dart (hunk 3 above)
Comment (staats-google) [inline dart/async/lib/webdriver_sync.dart]: The async => ensures that whatever is returned is a Future.

**Pull request merged**